src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectWorkspaceState.cs

```diff
@@ -9,13 +9,22 @@
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.Extensions.Internal;
+using Newtonsoft.Json;
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
 {
     public sealed class ProjectWorkspaceState : IEquatable<ProjectWorkspaceState>
     {
         public static readonly ProjectWorkspaceState Default = new ProjectWorkspaceState(Array.Empty<TagHelperDescriptor>(), LanguageVersion.Default);
 
+        public ProjectWorkspaceState(
+            IReadOnlyCollection<TagHelperDescriptor> tagHelpers,
+            LanguageVersion csharpLanguageVersion)
+            : this (tagHelpers.ToList(), csharpLanguageVersion)
```

Comment:
Is this ToList something that we'll plumb out later? I can't imagine that being a list can be THAT important vs a Collection?

Reply:
Unfortunately not. The `ProjectWorkspaceState` is the DTO that the rest of the Razor tooling system pulls from / expects to be an `IReadOnlyList`. Now that being said we could probably optimize this a little bit more because on the devenv side we use lists instead of HashSets meaning we could probably attempt a cast here and in practice it'd always succeed saving us the allocation!